kola/harness.go

```diff
@@ -367,11 +368,26 @@ func RunTests(patterns []string, channel, offering, pltfrm, outputDir string, ss
 
 	if !skipGetVersion {
 		plog.Info("Creating cluster to check semver...")
+
 		version, err := getClusterSemver(flight, outputDir)
 		if err != nil {
 			plog.Fatal(err)
 		}
 
+		// If the version is > 3033, we can safely use user-data instead of custom-data for
+		// provisioning the instance on Azure.
+		if version.Compare(semver.Version{Major: 3034}) == 1 && pltfrm == "azure" {
+			// Using reflection is a bit hacky, but it seems to be the only way to
+			// access the field we want to set.
+			f := reflect.ValueOf(flight).Elem()
+			api := f.FieldByName("Api")
+			opts := api.Elem().FieldByName("Opts")
+			userData := opts.Elem().FieldByName("UseUserData")
+			if userData.CanSet() {
+				userData.SetBool(true)
+			}
```

Comment:
```suggestion
			} else { some error }
```
When would `CanSet` be false?

Reply:
In theory never as the fields are known and exported. Since there is not even an error handling, let's drop the if condition.